jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFMetadataProcessor.java

```diff
@@ -23,6 +24,30 @@ public class JSFMetadataProcessor implements DeploymentUnitProcessor {
     private static final String DISALLOW_DOCTYPE_DECL = "com.sun.faces.disallowDoctypeDecl";
     private static final String LAZY_BEAN_VALIDATION_PARAM = "com.sun.faces.enableLazyBeanValidation";
 
+
+    private static final int defaultBufferSize;
+
+    // This is copied from org.wildfly.extension.undertow.ByteBufferPoolDefinition to come up with a decent default for
+    // the jakarta.faces.FACELETS_BUFFER_SIZE property. We calculate this because the default is 1024, which is very
+    // small, https://jakarta.ee/specifications/faces/4.0/jakarta-faces-4.0.html#a6088. Per the spec we could use -1
+    // as the default, but Mojarra does not currently support that. Once https://github.com/eclipse-ee4j/mojarra/issues/5262
+    // is resolved, and we can upgrade, we should be able to default to -1.
```

Comment:
@jamezp Should we create a WFLY issue to track defaulting to -1 once the fix for https://github.com/eclipse-ee4j/mojarra/issues/5262 is available in a release?

Reply:
@fjuma That's not a bad idea, https://issues.redhat.com/browse/WFLY-18174.

Reply:
Thanks @jamezp!